### PR TITLE
transpose solution

### PR DESCRIPTION
**Pull request by VGGatGitHub**
solution for the transpose exercise

### CODE REVIEW

**Comment by VGGatGitHub** [comment]
transpose solution update

File: 2020-06/transpose/transpose.py [hunk 1]
```diff
@@ -1,2 +1,44 @@
...
+                    text!=in_data[0] and out[i][0:1] !="."
+                    and out[i-1][-1:-2:-1] !="." and len(out[i-1])!=1 ):
+                        out[i]+=" "
+                    
+    
+        output+=out[i]
+        if i < n-1 :
+            output+='\n'
+    
+    if 1==2: #VGG print debuging
```
Comment (bwestover) [inline 2020-06/transpose/transpose.py:37]: I like that `1==2` flag. Change one character and you enable all the print statements below :heart:

File: 2020-06/transpose/transpose.py [hunk 2]
```diff
@@ -1,2 +1,44 @@
 def transpose(lines):
+
+    n_text=len(lines) #VGG number of letters in the full input set
+
+    in_data=lines.split("\n")
+    n_lines=len(in_data) #VGG number of text lines
+    
+    n=max([len(text) for text in in_data])
```
Comment (bwestover) [inline 2020-06/transpose/transpose.py:8]: You're grabbing the length of the longest line! Nice :+1:

File: 2020-06/transpose/transpose.py [hunk 3]
```diff
@@ -1,2 +1,44 @@
...
+
+                #VGG fixing the test_triangle
+                if len(out[i])>1 and len(out[i].strip())==0:
+                    out[i]+=" "
+
+                #VGG fixing the test_mixed_line_length
+                if len(out[i]) ==1 and (
+                    text!=in_data[0] and out[i][0:1] !="."
+                    and out[i-1][-1:-2:-1] !="." and len(out[i-1])!=1 ):
+                        out[i]+=" "
```
Comment (bwestover) [inline 2020-06/transpose/transpose.py:30]: This was a test I struggled to pass as well. I believe this is fixing the left-padding in the case where we get a long line, then a short line then a long line again, but I'm struggling to interpret this conditional statement :flushed: (it's pretty complicated).

I'm going to merge this in shortly, but if you get a second can you shed some light on how this is doing what it's doing?

**Review by bwestover — APPROVED** [review]

**Pull request merged**